framework/include/functions/PiecewiseLinearBase.h

```diff
@@ -28,12 +28,13 @@ class PiecewiseLinearBase : public PiecewiseBase
 
   PiecewiseLinearBase(const InputParameters & parameters);
 
-  virtual void initialSetup() override;
-  virtual Real value(Real t, const Point & pt) const override;
-  virtual Real timeDerivative(Real t, const Point & pt) const override;
-  virtual Real integral() const override;
-  virtual Real average() const override;
-  virtual void setData(const std::vector<Real> & x, const std::vector<Real> & y) override;
+  void initialSetup() override;
+  Real value(Real t, const Point & p) const override;
+  Real timeDerivative(Real t, const Point &) const override;
+  RealGradient gradient(Real, const Point & p) const override;
+  Real integral() const override;
+  Real average() const override;
+  void setData(const std::vector<Real> & x, const std::vector<Real> & y) override;
```

Comment:
I think these should remain virtual for clarity. If you don't want derived classes to override, then use the `final` keyword.

Reply:
Virtual is implied when the override keyword is used.

Reply:
Sooo, you prefer the virtual keyword for clarity? My personal preference is to avoid redundant syntax consistently, but I can add it back in if you wan, @zachmprince